libexec/rtld-elf/rtld_c18n.c

```diff
@@ -479,6 +479,8 @@ allocate_rstk_impl(unsigned index)
 }
 
 struct trusted_frame {
+	void *fp;
```

Comment:
This structure is very Morello specific and probably should be defined in rtld_machdep.h.  Especially if the notion is to align with the normal frame layout of the underlying architecture, different ABIs for passing args on different architectures, etc.

Reply:
@dstolfa It would be nice if the unwinder could also import this struct instead of hard-coding the offsets. Could you easily reach into RTLD to include header files from libunwind? (I assume adding a `-I` path in some Makefiles would do?)

Reply:
I'm not sure. I'm a little weary on including FreeBSD-specific headers for these kinds of things instead of having libunwind treat it as an ABI rather than a platform-dependent RTLD. Note that the current patches for libunwind don't include any FreeBSD-specific headers and instead just work under the assumption that this is the "CHERI compartmentalization ABI".